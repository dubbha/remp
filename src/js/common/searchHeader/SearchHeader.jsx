import React from 'react';
import PropTypes from 'prop-types';
import Logo from '../logo';
import SearchToolbar from './searchToolbar';
import ErrorBoundary from '../errorBoundary';
import './style.sass';

const SearchHeader = ({
  query,
  onSearch,
  onQueryChange,
  searchBy,
  onSearchByChange,
  searchByParams,
}) => (
  <header className="search">
    <ErrorBoundary>
      <Logo />
      <div className="search__header">find your movie</div>
      <form
        autoComplete="off"
        onSubmit={onSearch}
      >
        <div className="search__container">
          <input
            type="text"
            className="search__input"
            name="search"
            value={query}
            onChange={onQueryChange}
          />
          <div className="search__icon" />
        </div>
        <SearchToolbar
          searchBy={searchBy}
          onSearchByChange={onSearchByChange}
          searchByParams={searchByParams}
        />
      </form>
    </ErrorBoundary>
  </header>
);

SearchHeader.defaultProps = {
  query: '',
};

SearchHeader.propTypes = {
  query: PropTypes.string,
  onSearch: PropTypes.func.isRequired,
  onQueryChange: PropTypes.func.isRequired,
  searchBy: PropTypes.string.isRequired,
  onSearchByChange: PropTypes.func.isRequired,
  searchByParams: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default SearchHeader;
