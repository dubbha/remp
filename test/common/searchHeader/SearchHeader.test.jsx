import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchHeader from 'common/searchHeader';

describe('SearchHeader', () => {
  let props;

  beforeEach(() => {
    props = {
      searchBy: 'director',
      searchByParams: ['title', 'director'],
      onSearch: jest.fn(),
      onSearchByChange: jest.fn(),
      onQueryChange: jest.fn(),
    };
  });

  it('should render successfully', () => {
    const wrapper = shallow(<SearchHeader {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should pass the query string if available', () => {
    props.query = 'QUERY';
    const wrapper = shallow(<SearchHeader {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
