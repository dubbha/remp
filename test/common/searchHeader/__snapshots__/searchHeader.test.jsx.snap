// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeader should pass the query string if available 1`] = `
<header
  className="search"
>
  <ErrorBoundary>
    <Logo />
    <div
      className="search__header"
    >
      find your movie
    </div>
    <form
      autoComplete="off"
      onSubmit={[Function]}
    >
      <div
        className="search__container"
      >
        <input
          className="search__input"
          name="search"
          onChange={[Function]}
          type="text"
          value="QUERY"
        />
        <div
          className="search__icon"
        />
      </div>
      <SearchToolbar
        onSearchByChange={[Function]}
        searchBy="director"
        searchByParams={
          Array [
            "title",
            "director",
          ]
        }
      />
    </form>
  </ErrorBoundary>
</header>
`;

exports[`SearchHeader should render successfully 1`] = `
<header
  className="search"
>
  <ErrorBoundary>
    <Logo />
    <div
      className="search__header"
    >
      find your movie
    </div>
    <form
      autoComplete="off"
      onSubmit={[Function]}
    >
      <div
        className="search__container"
      >
        <input
          className="search__input"
          name="search"
          onChange={[Function]}
          type="text"
          value=""
        />
        <div
          className="search__icon"
        />
      </div>
      <SearchToolbar
        onSearchByChange={[Function]}
        searchBy="director"
        searchByParams={
          Array [
            "title",
            "director",
          ]
        }
      />
    </form>
  </ErrorBoundary>
</header>
`;
